src/sio_client.h

```diff
@@ -31,7 +36,7 @@ namespace sio
         
         typedef std::function<void(std::string const& nsp)> socket_listener;
         
-        client();
+        client(client_options const& options);
```

Comment:
```suggestion
        client(client_options const& options);
        client();
```

Do not remove the old constructor as this is backwards incompatible.